test/endpoint_testrunner.py

```diff
@@ -0,0 +1,238 @@
+                        })
+
+    return image_definitions
+
+
+def execute_endpoint_test(framework_name, image_definition, account_id, sagemaker_region, registry, region):
+    
+    repository = image_definition.get("repository")
+    framework_version = image_definition.get("framework_version")
+    processor = image_definition.get("processor")
+    python_version = image_definition.get("py_version")
+    tag = f"{framework_version}-{processor}-{python_version}"
+
+    retries = "" # " --reruns 2"
+    instance_type = "p2.xlarge"
+    python_version = ("2" if "27" in python_version else
+                      "3" if "36" in python_version else
+                      python_version.lstrip("py"))
+
+    test_location = os.path.join("test", "sagemaker_endpoint_tests", framework_name)
+
+    ctx = Context()
+    with ctx.cd(test_location):
+        run_out = ctx.run(f"pytest -rs{retries} test_endpoint.py "
+            f"--account-id {account_id} "
+            f"--region {region} "
+            f"--registry {registry} "
+            f"--repository {repository} "
+            f"--instance-type ml.{instance_type} "
+            f"--framework-version {framework_version} "
+            f"--sagemaker-region {sagemaker_region} "
+            f"--py-version {python_version} "
+            f"--tag {tag}",
+            warn=True, echo=True
+        )
+    return run_out.ok, run_out.stdout
+
+
+def generate_image_uri(image, registry, region):
+    repository = image.get("repository")
+    framework_version = image.get("framework_version")
+    processor = image.get("processor")
+    py_version = image.get("py_version")
+    domain_suffix = ".cn" if region in ("cn-north-1", "cn-northwest-1") else ""
+    return f"{registry}.dkr.ecr.{region}.amazonaws.com{domain_suffix}/{repository}:{framework_version}-{processor}-{py_version}"
+
+
+def run_framework_tests(framework, images, canary_account_id, sagemaker_region, registry, region):
+    if not images:
+        generate_report(REPORT_XML, "{framework} endpoint test", "sagemaker-endpoint", "No framework iaage to test.")
+        return
+    
+    """
+    # enable if there are sufficient instances in region
+    # to run process in parallel
+    results = []
+    pool_number = len(images)
+    with Pool(pool_number) as p:
+        results = p.starmap(
+            execute_endpoint_test,
+            [[framework, images[i], canary_account_id, sagemaker_region, registry, region] for i in range(pool_number)]
+        )
+    """
+    # run tests sequentially
+    for image in images:
+        test_status, test_logs = execute_endpoint_test(framework, image, canary_account_id, sagemaker_region, registry, region)
+        if not test_status:
+            image_uri = generate_image_uri(image, registry, region)
+            LOGGER.error(f"Endpoint test failed for image {image_uri}")
+            generate_report(REPORT_XML, f"{framework} endpoint test", "sagemaker-endpoint", "", test_logs)
```

Comment:
you shouldn't need to generate the report separately if you use the --junitxml pytest flag